sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1167,24 +1167,24 @@ def test_radius_neighbors_graph_sparse(seed=36):
 
 def test_neighbors_badargs():
     # Test bad argument values: these should all raise ValueErrors
-    with pytest.raises(ValueError):
-        neighbors.NearestNeighbors(algorithm='blah')
-
     X = rng.random_sample((10, 2))
     Xsparse = csr_matrix(X)
     X3 = rng.random_sample((10, 3))
     y = np.ones(10)
 
+    with pytest.raises(ValueError):
+        neighbors.NearestNeighbors(algorithm='blah').fit(X)
+
     for cls in (neighbors.KNeighborsClassifier,
                 neighbors.RadiusNeighborsClassifier,
                 neighbors.KNeighborsRegressor,
                 neighbors.RadiusNeighborsRegressor):
         with pytest.raises(ValueError):
-            cls(weights='blah')
+            cls(weights='blah').fit(X, y)
         with pytest.raises(ValueError):
-            cls(p=-1)
+            cls(p=-1).fit(X, y)
         with pytest.raises(ValueError):
-            cls(algorithm='blah')
+            cls(algorithm='blah').fit(X, y)
```

Comment:
To make use that the `ValueError` is not raised in `__init__`:

```python
est = Cls(weights='blah')
with pytest.raises(ValueError):
    est.fit(X, y)
```

And similar for the other changes.

Reply:
Done!